sqlboiler/graph/templates/singleton/helper.tpl

```diff
@@ -56,12 +56,12 @@ func QueryModPagination(sinceID *graphql.ID, pageNum, pageSize *int32) []qm.Quer
 // QueryModSearch returns a list of search QueryMod based on the struct values
 func QueryModSearch(input interface{}) []qm.QueryMod {
   mods := []qm.QueryMod{}
-  if reflect.ValueOf(input).IsNil() {
```

Comment:
this was added in https://github.com/choonkeat/hellocrud/commit/792e7a6c9bac42c4c6bd186a1f0493efe5d12a63#diff-a3e23cbdf329a381fbbbb9fce2aaa1f6

do you want to instead

``` go
input == nil || reflect.ValueOf(input).IsNil()
```

Reply:
the argument to `QueryModSearch` is a pointer `*search{{$modelName}}Input`; can be `nil`

``` go
func (r *Resolver) Search{{$modelNamePlural}}(ctx context.Context, args struct {
	// ...
	Input *search{{$modelName}}Input
}) ({{$modelNamePlural}}Collection, error) {
	// ...
	mods = append(mods, QueryModSearch(args.Input)...)
```

Reply:
And didn't know that `interface == nil` won't work in this scenario due to https://github.com/golang/go/issues/1288

Reply:
Oh, it's because of this: 
https://stackoverflow.com/questions/29138591/hiding-nil-values-understanding-why-golang-fails-here/29138676#29138676